flocker/node/_docker.py

```diff
@@ -457,6 +465,20 @@ def _extract_error(failure):
         d.addErrback(_extract_error)
         return d
 
+    def _blocking_wait(self, timeout, is_ready, arg):
+        """
+        Wait until a condition is satisfied.
+
+        :param function is_ready: Function testing whether condition is
+            satisfied.
+        :param unicode arg: An argument to pass to the is_ready function.
+        """
+        time_limit = time() + timeout
+        while not is_ready(arg):
+            if time() > time_limit:
+                raise RuntimeError('Timeout > %s seconds' % timeout)
```

Comment:
minor point, `RuntimeError` is for errors *in* the runtime (ie, CPython) rather than errors *at* runtime.